src/main/java/de/focusshift/zeiterfassung/absence/AbsenceServiceImpl.java

```diff
@@ -136,17 +164,74 @@ private static void doFromUntil(LocalDate from, LocalDate toExclusive, Consumer<
         }
     }
 
-    private static Absence toAbsence(AbsenceWriteEntity entity, ZoneId zoneId) {
+    private Stream<Absence> toAbsences(List<AbsenceWriteEntity> absenceEntities) {
+
+        final ZoneId zoneId = userSettingsProvider.zoneId();
+
+        final Map<Long, AbsenceType> absenceTypeBySourceId = findAbsenceTypes(absenceEntities).stream()
+            .collect(toMap(AbsenceType::sourceId, identity()));
+
+        final Function<Locale, String> sickLabel = new CachedFunction<>(this::sickLabel);
+
+        return absenceEntities.stream()
+            .map(entity -> toAbsence(entity, zoneId, absenceTypeBySourceId::get, sickLabel));
+    }
+
+    private String sickLabel(Locale locale) {
+        return messageSource.getMessage("absence.type.category.SICK", null, locale);
+    }
+
+    private Absence toAbsence(AbsenceWriteEntity entity, ZoneId zoneId, LongFunction<AbsenceType> absenceTypeBySourceIdSupplier, Function<Locale, String> sickLabelSupplier) {
+
+        final AbsenceType absenceType;
+        if (entity.getType().getCategory().equals(SICK)) {
+            absenceType = new AbsenceType(SICK, null, sickLabelSupplier, ABSENCE_SICK_COLOR);
+        } else {
+            final AbsenceTypeEntityEmbeddable type = entity.getType();
+            absenceType = absenceTypeBySourceIdSupplier.apply(type.getSourceId());
+            if (absenceType == null) {
+                // TODO how to handle `absenceType == null`?
```

Comment:
was bedeutet eine Abwesenheit ohne der Zeiterfassung bekannte Abwesenheitsart? Wann passiert das?

U.a. ist hier die Reihenfolge von messages relevant (Abwesenheit wird gelesen bevor in der Zeiterfassung die Abwesenheitsart bekannt ist). 
- in diesem Fall würde ich die Abwesenheit hier ignorieren

--> soll eine Abwesenheit mit fehlender Abwesenheitsart generell ignoriert werden?

Reply:
Wenn man die Reihenfolge als mögliche Quelle nimmt, würde ich auch hier ignorieren (warn log), aber generell persistieren, weil man ja irgendwann wieder in einen konsistenten Zustand kommen sollte